src/algebraic_geometry/prime_spectrum/maximal.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2022 David Kurniadi Angdinata. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata
+-/
+
+import algebraic_geometry.prime_spectrum.basic
+
+/-!
+# Maximal spectrum of a commutative ring
+
+The maximal spectrum of a commutative ring is the type of all maximal ideals.
+It is naturally a subset of the prime spectrum endowed with the subspace topology.
+
+## Main definitions
+
+* `maximal_spectrum R`: The maximal spectrum of a commutative ring `R`,
+  i.e., the set of all maximal ideals of `R`.
+
+## Implementation notes
+
+The Zariski topology on the maximal spectrum is defined as the subspace topology induced by the
+natural inclusion into the prime spectrum to avoid API duplication for zero loci.
+-/
+
+noncomputable theory
+open_locale classical
+
+universes u v
+
+variables (R : Type u) [comm_ring R]
+
+/-- The maximal spectrum of a commutative ring `R` is the type of all maximal ideals of `R`. -/
+@[ext, nolint has_nonempty_instance] structure maximal_spectrum :=
```

Comment:
I think the nolint could be removed if there is `instance [nontrivial R] : nonempty (maximal_spectrum R)`.